extensions/functions_geometry.yaml

```diff
@@ -0,0 +1,68 @@
+%YAML 1.2
+---
+scalar_functions:
+  -
+    name: "isvalid"
+    description: >
+      Return true if the input geometry is a valid 2D geometry.
+
+      For 3 dimensional and 4 dimensional geometries, the validity is still only tested
+      in 2 dimensions.
+    impls:
+      - args:
+          - value: geometry
+            name: geom
+        return: boolean
+  -
+    name: "collection_extract"
+    description: >
+      Given the input geometry collection, return a homogenous multi-geometry.  All geometries
+      in the multi-geometry will have the same dimension.
+
+      If type is not specified, the multi-geometry will only contain geometries of the highest
+      dimension.  If type is specified, the multi-geometry will only contain geometries
+      of that type.  If there are no geometries of the specified type, an empty geometry
+      is returned.  Only points, linestrings, and polygons are supported.
+
+      Type numbers:
+      1 == POINT
+      2 == LINESTRING
+      3 == POLYGON
+    impls:
+      - args:
+          - value: geometry
+            name: geom_collection
+        return: geometry
+      - args:
+          - value: geometry
+            name: geom_collection
+          - value: i8
+            name: type
+        return: geometry
+  -
+    name: "flip_coordinates"
+    description: >
+      Return a version of the input geometry with the X and Y axis flipped.
```

Comment:
This appears to be another 2D operation.  We should probably state what happens to the other dimensions and/or rename the function to clarify its 2D nature.